pkg/controller/migplan/migplan_controller.go

```diff
@@ -136,6 +136,24 @@ func (r *ReconcileMigPlan) Reconcile(request reconcile.Request) (reconcile.Resul
 			return reconcile.Result{}, err
 		}
 	}
+
+	// plan is closed, ensure resource cleanup
+	if plan.Status.HasCondition(Closed) {
```

Comment:
I think it will make the code easier to grok if we're only dealing with one err at a time where possible. I think the below is accomplishes something similar to this block, assuming it's OK to mark a plan as not ready (which seems sane to me) when we know it is closed and are deleting its registries.

```
// Plan is closed, ensure resource cleanup
if plan.Status.HasCondition(Closed) {
    
    // Set plan as not ready since it is marked as closed
    plan.Status.SetReady(false, ReadyMessage)
    err := r.Update(context.TODO(), plan)
    if err != nil {
        if errors.IsConflict(err) {
            return reconcile.Result{Requeue: true}, nil
        }
        return reconcile.Result{}, err
    }


    // Migration Registry cleanup
    err = r.ensureMigRegistriesDelete(plan)
    if err != nil {
        if errors.IsConflict(err) {
            return reconcile.Result{Requeue: true}, nil
        }
        return reconcile.Result{}, err
    }
}
```

Reply:
The thing we need to confirm is that since, with this modification, we may leave this reconcile loop without deleting its registries, we need to make sure that the `SetReady(false)`  doesn't somehow clear out the `Closed` condition. One thing the current version of this code does is to ensure that it attempts to both `SetReady(false)` and delete the registres. If possibly splitting them up into separate reconcile calls is OK, then this modification should be fine.

Reply:
I recommend moving this block into a top level `ensureClosed()` in a spearte .go file.  This well keep the main reconcile() cleaner especially when the storage clean up is added.